### PR TITLE
Fix ofborg eval

### PR DESCRIPTION
**Pull request by infinisil**
###### Motivation for this change
See https://github.com/NixOS/nixpkgs/pull/66493

I'll also make a PR to make this change permanent to the haskell-updates branch. Edit: Done in https://github.com/NixOS/nixpkgs/commit/1aa604fcda92949d045038ecb341042a82fa960e and https://github.com/NixOS/nixpkgs/commit/141ce43c840a5750a12aadbca2d0965dc2634e61

cc @grahamc @worldofpeace @timokau @peti

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix
Comment (grahamc) [inline pkgs/development/haskell-modules/hackage-packages.nix]: This might fix eval, but is it actually broken?

Comment (Infinisil) [reply pkgs/development/haskell-modules/hackage-packages.nix]: See https://github.com/NixOS/nixpkgs/pull/66493#issuecomment-520264858

Comment (timokau) [reply pkgs/development/haskell-modules/hackage-packages.nix]: Why not keep the `hydraPlatforms`?

**Comment by Infinisil** [comment]
Permanent fix pushed in 1aa604fcda9

File: pkgs/development/haskell-modules/hackage-packages.nix
Comment (timokau) [reply pkgs/development/haskell-modules/hackage-packages.nix]: Explained in https://github.com/NixOS/nixpkgs/pull/66493#issuecomment-520264858

Comment (Infinisil) [reply pkgs/development/haskell-modules/hackage-packages.nix]: I just did what a package not marked as broken would generate

**Review by worldofpeace — DISMISSED** [review]
Ran `./maintainers/scripts/eval-release.sh` :+1: 
(I think that's the right script to run)

**Comment by worldofpeace** [comment]
hmm there's another according to ofborg
```
Failed to evaluate http-download-0.1.0.0: «broken»: is marked as broken
```

**Comment by Infinisil** [comment]
Kinda suspected there might be more failures.. lemme see how to fix

**Comment by Infinisil** [comment]
Seems that the same thing happened for pretty much everything @peti fixed the build of

**Comment by Infinisil** [comment]
I think these are all though, pretty sure eval should succeed now, let's see

**Comment by grahamc** [comment]
Unfortunately ofborg will always fail the PR because master fails. I've intended on fixing this :/

**Comment by worldofpeace** [comment]
> Unfortunately ofborg will always fail the PR because master fails. I've intended on fixing this :/

Ah I see, though it did seem to progress.

---

The next eval failure is unrelated to this in 
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/syncthing-gtk/default.nix

Because of the `gdk-pixbuf` rename.

**Comment by Infinisil** [comment]
Ideally ofborg would evaluate both master and the (merged) PR, then it can detect:
- master broken, PR broken -> Eval was broken and PR doesn't fix it, hold on and evaluate again once a PR that fixes eval is merged
- master broken, PR works -> PR fixes eval, good to merge, trigger above
- master works, PR broken -> PR breaks eval, blocks merging
- master works, PR works -> PR doesn't break eval, good to merge

**Comment by Infinisil** [comment]
Ah darn, a pixbuf error.. https://github.com/NixOS/nixpkgs/pull/61876 and https://github.com/NixOS/cabal2nix/pull/424

**Comment by worldofpeace** [comment]
@Infinisil It appears that's a new addition since yesterday. https://github.com/NixOS/nixpkgs/pull/66495#issuecomment-520267336

**Comment by Infinisil** [comment]
Seems that hackage2nix version 2.14.4 [was used](https://github.com/NixOS/nixpkgs/pull/66468/commits/44afe302454b4699ac056d7e68b861c6656fea43), not master (which I thought @peti used), so another manual fix like https://github.com/NixOS/nixpkgs/pull/65487 is needed

**Comment by Infinisil** [comment]
Wait no

**Comment by Infinisil** [comment]
Right, sorry @worldofpeace I didn't really look at your comment close enough, you're right, ignore what I said

**Comment by Infinisil** [comment]
Ah, https://github.com/NixOS/nixpkgs/pull/66375 is the cause, merged ~30 minutes ago

**Comment by worldofpeace** [comment]
> Right, sorry @worldofpeace I didn't really look at your comment close enough, you're right, ignore what I said

:smile: My thought process can go like this as well.

**Comment by Infinisil** [comment]
All green, I'll merge

**Pull request merged**
